### PR TITLE
EVG-12948 Improve process cleanup

### PR DESCRIPTION
**Pull request by ybrill**
For Linux and macOS hosts, if a process started by `shell.exec` or `subprocess.exec` starts a child process without the child process inheriting the environment of its parent `killProcs` will not cleanup the child process. This PR adds two mitigations:
- Give users an environment variable that's easy to include in the environments of child processes they start. Setting the `IN_EVERGREEN` variable will flag a process to be killed.
- Kill a process whose executable is located in the working dir of the task. The implementation is platform dependent:
  - Linux checks if the working dir is a prefix of the process's `/proc/<PID>/exe` symlink.
  - macOS doesn't provide a way to find the path to the executable so it checks if the working dir is a prefix to the command.

This doesn't apply to Windows because its cleanup works differently. The original process is added to a job object and, by default, [children are also associated with their parents' job](https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects?redirectedfrom=MSDN#managing-processes-in-jobs) depending on how the children are created.

### CODE REVIEW

File: util/subtree_darwin.go [hunk 1]
```diff
@@ -46,17 +46,23 @@ func cleanup(key string, logger grip.Journaler) error {
...
-			pidAsInt, err := strconv.Atoi(pid)
-			if err != nil {
-				logger.Errorf("cleanup failed to convert from string to int: %v", err)
-				continue
-			}
-			pidsToKill = append(pidsToKill, pidAsInt)
+		if pid == myPid {
+			continue
+		}
+		if !envHasMarkers(key, env) && !strings.HasPrefix(command, workingDir) {
```
Comment (bsamek) [inline util/subtree_darwin.go:55]: Is there a risk that `workingDir` will be somehow unset and the agent kills way more than it intends to? I wonder if there's some sanity check we can add somewhere to reduce the risk of that. Maybe just check that `workingDir` is not the empty string?

File: util/subtree_linux.go [hunk 2]
```diff
@@ -79,37 +80,35 @@ func listProc() ([]int, error) {
 			if err != nil {
 				continue
 			}
+
+			if pid == myPid {
+				continue
+			}
+
+			if !processHasMarkers(pid, key, logger) && !executableInWorkingDir(pid, workingDir, logger) {
```
Comment (bsamek) [inline util/subtree_linux.go:88]: Same question about the working dir if it's the empty string.

**Review by bsamek — CHANGES_REQUESTED** [review]
Do we have tests of the kill behavior?

File: util/subtree_darwin.go (hunk 1 above)
Comment (ybrill) [reply util/subtree_darwin.go]: done

File: util/subtree_linux.go (hunk 2 above)
Comment (ybrill) [reply util/subtree_linux.go:88]: caught the empty string case

**Comment by ybrill** [comment]
I added tests for getting the list of pids to kill

**Review by bsamek — APPROVED** [review]

**Pull request merged**